### PR TITLE
Extension to view MicroProfile config values in dev mode

### PR DESCRIPTION
**Pull request by hpehl**
PR for #1349. 

Adds a servlet in dev mode which lists the configured values for MicroProfile Configuration. The servlet is registered using the path `/config` by default and lists all properties of all config sources. The config sources are sorted descending by ordinal, the properties by name. If no config is defined an empty JSON object is returned.

There's no integration test in this PR. Since the servlet is only added in dev mode, `@QuarkusTest` does not work as expected. 

If necessary I can provide documentation.

### CODE REVIEW

File: extensions/config/deployment/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    <artifactId>quarkus-config-deployment</artifactId>
+    <name>Quarkus - Show Configuration - Deployment</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core-deployment</artifactId>
+        </dependency>
+        <dependency>
```
Comment (kenfinnigan) [inline extensions/config/deployment/pom.xml]: I would be inclined to go lower and just have it depend on vertx instead of resteasy

**Review by kenfinnigan — CHANGES_REQUESTED** [review]
I have a problem with the names being "config" for the config and "quarkus-config" for the extension.

To me that implies these bring with them the ability to configure an application when that functionality is already present.

Maybe a name like "config-viewer", not sure tbh

File: extensions/config/deployment/pom.xml (hunk 1 above)
Comment (hpehl) [reply extensions/config/deployment/pom.xml]: Good point. I'll adjust the dependencies.

**Comment by hpehl** [comment]
I agree about the naming issue. Since it just dumps the current configuration, maybe `quarkus-config-dump` is a better choice.

**Comment by kenfinnigan** [comment]
Maybe `quarkus-config-viewer`?

**Comment by hpehl** [comment]
Sorry my last commit introduced too many changes. I'll close this PR and re-submit into a new PR.

**Comment by hpehl** [comment]
Reopened as #5692

**Pull request closed without merging**